build.sh

```diff
@@ -112,7 +112,7 @@ prepare_native_build()
     # Generate version.c if specified, else have an empty one.
     __versionSourceFile=$__scriptpath/bin/obj/version.c
     if [ $__generateversionsource == true ]; then
-        $__scriptpath/Tools/corerun $__scriptpath/Tools/MSBuild.exe "$__scriptpath/build.proj" /t:GenerateVersionSourceFile /p:NativeVersionSourceFile=$__scriptpath/bin/obj/version.c /p:GenerateVersionSourceFile=true /v:minimal
```

Comment:
My gut feel is that passing all the unprocessed arguments to this msbuild invocation is going to be problematic or confusing long term.  I would expect the extra arguments I pass to build.sh to only flow to the msbuild for build.proj. Maybe we should teach build.sh and build.cmd about a special argument which specifies we are doing an official build and then passes the argument to msbuild here as necessary?

Interested in how other folks feel about this.